util/unifycr/src/unifycr-rm.c

```diff
@@ -169,30 +244,82 @@ static int read_resource_lsf(unifycr_resource_t *resource)
     return 0;
 }
 
+/**
+ * @brief
+ *
+ * @param resource
+ * @param state
+ *
+ * @return
+ */
+static int lsf_launch_daemon(unifycr_resource_t *resource,
+                             unifycr_runstate_t *state)
+{
+    return 0;
+}
+
+static _ucr_resource_manager_t resource_managers[] = {
+    { "none", &none_read_resource, &none_launch_daemon },
+    { "pbs", &pbs_read_resource, &pbs_launch_daemon },
+    { "slurm", &slurm_read_resource, &slurm_launch_daemon },
+    { "lsf", &lsf_read_resource, &lsf_launch_daemon },
+};
+
 int unifycr_read_resource(unifycr_resource_t *resource)
 {
-    int ret = -1;
     char *str = NULL;
 
+    /*
+     * "do not use assignement in if condition"
+     * so, here we use goto instead of if .. else ... any better way?
+     */
+
     str = getenv("PBS_JOBID");
     if (str) {
-        ret = read_resource_pbs(resource);
+        resource->rm = UNIFYCR_RM_PBS;
         goto out;
     }
 
     str = getenv("SLURM_JOBID");
     if (str) {
-        ret = read_resource_slurm(resource);
+        resource->rm = UNIFYCR_RM_SLURM;
         goto out;
     }
 
     str = getenv("LSB_JOBID");
     if (str) {
-        ret = read_resource_lsf(resource);
+        resource->rm = UNIFYCR_RM_LSF;
         goto out;
     }
 
+    resource->rm = UNIFYCR_RM_NONE;
+
 out:
-    return ret;
+    return resource_managers[resource->rm].read_resource(resource);
+}
+
+/*
+ * TODO: currently it directly uses mpirun command in any cases.  The command
+ * should be:
+ * $ mpirun -n NUMNODES -np NUMNODES unifycrd
+ */
+static char *mpirun_newargv[] = {
+    "mpirun", "-n", NULL, "-np", NULL, BINDIR "/unifycrd", (char *) NULL
```

Comment:
It would be useful to be able to specify the path to unifycrd in the configuration file, and default to the BINDIR location if unspecified, rather than hard coding it here.

Reply:
@nedbass agreed. i will make changes accordingly.